component/identity/src/main/java/org/exoplatform/services/organization/idm/GroupDAOImpl.java

```diff
@@ -563,17 +571,31 @@ public Collection<Group> findGroupsOfUserByKeyword(String user, String keyword,
         return null;
       }
       Collection<org.picketlink.idm.api.Group> allGroups = new HashSet<org.picketlink.idm.api.Group>();
-      try {
-        orgService.flush();
-        allGroups = getIdentitySession().getRelationshipManager().findRelatedGroups(user, groupType, identitySearchCriteria);
-      } catch (Exception e) {
-        // TODO:
-        handleException("Identity operation error: ", e);
+      Identity identity = ConversationState.getCurrent().getIdentity();
```

Comment:
You can't change the behavior of this method here, it can be already used by other services that need the list of groups of a user no matter if he is an admin or not, your needs should be handled in the suggestion service, not in the DAO layer